### DISCUSSION TITLE
Improvements of Rewatch feature support through the API and website

### DISCUSSION THREAD

**Discussion by kevincador** [trakt/trakt-api#269] [Q&A]
Some things are missing from the API:

- A way to set the `reset_at` via the API
- It would be really cool if the `next_episode` from the progress endpoint could be the next one that needs to be rewatched. I currently work around this but then the actions app are limited.
- A property that says you need to rewatch and some counts would also help (adding properties shouldn't break any apps/systems relying on the API).
- At least the trakt identifier would be helpful in the progress endpoint (but that's another discussion).

Some things that I feel are missing even on the website:

- Removing/deleting the `reset_at` e.g. if I made a mistake
- Instead of setting the `reset_at` via a date, it would be easier to start rewatching from a specific episode onward
- I'm missing a way to filter out everything I'm not rewatching from the progress page

And one thing that seems to be buggy:

- When I start a rewatch, the progress endpoint is not updated directly (may be a cache issue). If I mark something watch, the progress is correctly updated and my app is updated but it isn't updated after just setting the `reset_at` from the website.

**Comment by rectifyer** [comment]
Some good suggestions here. We don't currently allow resetting via the API since this is a VIP only feature. However, I think it might make sense to add this and return an error code `426` for non VIP members.

You can get a list of all reset shows using https://trakt.docs.apiary.io/#reference/users/hidden-items/get-hidden-items and using `progress_watched_reset` as the `section` parameter.

The [progress API](https://trakt.docs.apiary.io/#reference/shows/collection-progress/get-show-watched-progress) also includes the reset_at property. Per the docs on that call, you can re-calculate the progress for a show in app.

> If not null, the `reset_at` date is when the user started re-watching the show. Your app can adjust the progress by ignoring episodes with a `last_watched_at` prior to the `reset_at`.

On the website, you can undo any rewatching on the https://trakt.tv/settings/hidden page. I do agree we should have some additional filters on the progress page to view only rewatching and a way to filter them out.

That last issue does sound cache related, I'll need to look into that one deeper.

**Reply by kevincador** [reply]
Cool. It seems like the reset/rewatch is linked to the hidden. I didn't know that. I guess the documentation needs a refresh for this subject 😉. Does that mean that if is possible to add a reset/rewatch via the hidden API already?

I'm currently using the progress API to display the show in my "To Watch". The biggest issue I'm finding right now is that the `next_episode` is considered an episode in my parsing and the episodes in seasons in can't because they don't have an identifier (I'm parsing them differently). 

The undoing of a rewatch is pretty hidden 🙈 but once you know, you know.

Thanks for your feedback!

**Comment by rectifyer** [comment]
Still a work in progress, but we have the basic VIP only methods available to reset watched progress.

https://trakt.docs.apiary.io/#reference/shows/reset-watched-progress/reset-show-progress
https://trakt.docs.apiary.io/#reference/shows/reset-watched-progress/undo-reset-show-progress
